### PR TITLE
feature: support pod metadata for DataProcess and simplify script processor's API

### PR DESCRIPTION
**Pull request by TrafalgarZZZ**
### Ⅰ. Describe what this PR does
- Support pod metadata (like labels and annotations) for DataProcess
- Simplify script processor's API to allow users define raw shell/python/... scripts.

For example, you can use the following sample to run a DataProcess job which sleeps for 10 seconds:
```
apiVersion: data.fluid.io/v1alpha1
kind: DataProcess
metadata:
  name: process-test
spec:
  dataset:
    name: jfsdemo
    namespace: default
    mountPath: /data
  processor:
    script:
      image: nginx
      imageTag: latest
      command: ["bash"]
      script: |
        #!/bin/bash
        set -ex
        sha256sum /data/models/Old-Demo-6.mov > /data/models/256sum-result.txt
```

### Ⅱ. Does this pull request fix one issue?

#3391 

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: charts/fluid-dataprocess/common/templates/configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Release.Name }}-scripts
+  labels:
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+data:
+  preprocess.sh: {{- .Values.dataProcess.scriptProcessor.source | toYaml | indent 2}}
```
Comment (cheyang) [inline charts/fluid-dataprocess/common/templates/configmap.yaml:9]: Add return of line.

File: charts/fluid-dataprocess/common/templates/configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Release.Name }}-scripts
+  labels:
+    release: {{ .Release.Name }}
```
Comment (cheyang) [inline charts/fluid-dataprocess/common/templates/configmap.yaml:6]: I think the ownerReference may help a lot.

Comment (TrafalgarZZZ) [reply charts/fluid-dataprocess/common/templates/configmap.yaml:6]: Done

Comment (TrafalgarZZZ) [reply charts/fluid-dataprocess/common/templates/configmap.yaml:6]: Done.

**Review by cheyang — APPROVED** [review]
/lgtm
/approve

**Pull request merged**